### PR TITLE
Add stage channel methods 

### PR DESCRIPTION
**Pull request by ashwinvin**
### Summary
Add  stage channel methods for RESTClient and GuildBuilder

### Checklist

- [x] I have run `nox` and all the pipelines have passed.
- [x] I have made unittests according to the code I have added

### CODE REVIEW

**Comment by davfsa** [comment]
@ashwinvin Looks good. Could you mind adding testing for the rest endpoints. I will do the special endpoints ones in the future :)

**Comment by ashwinvin** [comment]
Sure!

**Review by davfsa — APPROVED** [review]

File: tests/hikari/impl/test_rest.py [hunk 1]
```diff
@@ -2648,6 +2648,39 @@ async def test_create_guild_voice_channel(self, rest_client):
...
+            "general",
+            position=1,
+            user_limit=60,
+            bitrate=64,
+            permission_overwrites=[overwrite1, overwrite2],
+            category=category_channel,
+            region="Doge Moon",
+            reason="When doge == 1$",
+        )
+        assert returned == channel
```
Comment (FasterSpeeding) [inline tests/hikari/impl/test_rest.py:2670]: couldn't `is` be used here to check?

Comment (FasterSpeeding) [reply tests/hikari/impl/test_rest.py:2670]: oh didn't notice that, might as well just leave that style then ig

**Review by FasterSpeeding — APPROVED** [review]

File: tests/hikari/impl/test_rest.py (hunk 1 above)
Comment (davfsa) [reply tests/hikari/impl/test_rest.py:2670]: These should def be using `is`. I will have to fix the other tests

**Pull request merged**